File: FrameWorkBaseProject/FrameWorkBaseProject/ViewController.m
//
//  ViewController.m
//  FrameWorkBaseProject
//
//  Created by zhangliang on 14-10-16.
//  Copyright (c) 2014年 com.yourcompany.orgnanization. All rights reserved.
//

#import "ViewController.h"
#import <FrameworkTest/FrameworkTest.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    ObjectOne * object = [[ObjectOne alloc] init];
    [object sayHello];
    [object baby];
    
    
    ObjectTwo * objectTwo = [[ObjectTwo alloc] init];
    [objectTwo saySelf];
    NSLog(@"%d", [objectTwo sum:3 and:4]);
    NSLog(@"%@", [objectTwo appStr:@"123456" andSecStr:@"7890"]);
    
    
    NSLog(@"%d", [self sum:3 and:4]);
    NSLog(@"%@", [self appStr:@"123456" andSecStr:@"7890"]);
    
    NSLog(@"hello, baby");
}

- (int)sum:(int)num and:(int)num2
{
    int sum = 0;
    sum = num + num2;
    return sum;
}

- (NSString *)appStr:(NSString *)firStr andSecStr:(NSString *)secStr
{
    NSString * str = [[NSString alloc] initWithFormat:@"%@%@", firStr, secStr];
    return str;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: AdaptiveLayout/AdaptiveLayout/ViewController.swift
//
//  ViewController.swift
//  AdaptiveLayout
//
//  Created by zhangliang on 14-10-14.
//  Copyright (c) 2014年 com.yourcompany.orgnanization. All rights reserved.
//

import UIKit

import LocalAuthentication

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        var authButton: UIButton = UIButton.buttonWithType(UIButtonType.System) as UIButton
        authButton.frame = CGRect(x: 100, y:UIScreen.mainScreen().bounds.size.height / 2, width: 100, height: 30)
        authButton.setTitle("Authenticate", forState: UIControlState.Normal)
        authButton.addTarget(self, action: Selector("addPassAction:"), forControlEvents: UIControlEvents.TouchUpInside)
        self.view.addSubview(authButton)
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)

    }
    
    func addPassAction(sender:UIButton!){
        
        var laContext = LAContext()
        laContext.localizedFallbackTitle = "不使用指纹"
        var authError : NSError?
        var errorReason = "请用指纹验证"
        
        if laContext.canEvaluatePolicy(LAPolicy.DeviceOwnerAuthenticationWithBiometrics, error: &authError){
            laContext.evaluatePolicy(LAPolicy.DeviceOwnerAuthenticationWithBiometrics, localizedReason: errorReason, reply: {
                (success, error) in
                if success {
                    var alertVC:UIAlertController = UIAlertController(title:"结果展示", message: "成功", preferredStyle: UIAlertControllerStyle.Alert)
                    var alertActionSure: UIAlertAction = UIAlertAction(title: "确定", style: UIAlertActionStyle.Default, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                    var alertActionCancel: UIAlertAction = UIAlertAction(title: "取消", style: UIAlertActionStyle.Cancel, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                    alertVC.addAction(alertActionSure)
                    alertVC.addAction(alertActionCancel)
                    self.presentViewController(alertVC, animated:true, completion: nil)
                }
                else{
                    
                    if error!.code == -2 {
                        var alertVC:UIAlertController = UIAlertController(title:"结果展示 " + "用户取消了指纹输入", message: "失败", preferredStyle: UIAlertControllerStyle.Alert)
                        var alertActionSure: UIAlertAction = UIAlertAction(title: "确定", style: UIAlertActionStyle.Default, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                        var alertActionCancel: UIAlertAction = UIAlertAction(title: "取消", style: UIAlertActionStyle.Cancel, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                        alertVC.addAction(alertActionSure)
                        alertVC.addAction(alertActionCancel)
                        self.presentViewController(alertVC, animated:true, completion: nil)
                    }else if error!.code == -3 {
                        println(error!.userInfo![NSLocalizedDescriptionKey])
                        var alertVC:UIAlertController = UIAlertController(title:"结果展示" + error!.userInfo!.description , message: "失败", preferredStyle: UIAlertControllerStyle.Alert)
                        var alertActionSure: UIAlertAction = UIAlertAction(title: "确定", style: UIAlertActionStyle.Default, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                        var alertActionCancel: UIAlertAction = UIAlertAction(title: "取消", style: UIAlertActionStyle.Cancel, handler: {action in alertVC.dismissViewControllerAnimated(true, completion: nil)})
                        alertVC.addAction(alertActionSure)
                        alertVC.addAction(alertActionCancel)
                        self.presentViewController(alertVC, animated:true, completion: nil)
                    }
                }
            })
        }
        else{
            var alert = UIAlertView(title: "Can not do authenticatation", message: authError!.description, delegate: nil, cancelButtonTitle: "Cancel")
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

